build.gradle

```diff
@@ -0,0 +1,190 @@
+plugins {
+	id 'java'
+	id 'war'
+	id 'jacoco'
+	id 'org.ajoberstar.grgit' version '4.1.1'
+}
+
+// TODO NEXT build script - how include built jar
+// TODO NEXT update dockerfile
+// TODO NEXT remove ant & makefile
+// TODO NEXT move to standard gradle locations
+// TODO NEXT run tests from Eclipse w/o specifying classpath manually & remove sourceSets & classpath
+// TODO NEXT use jitpack jars vs pulling from repo
+
+// TODO TEST DEPS TypeGeneratorTest has a hard coded list of jars it uses for tests
+//           Need to figure out a better way
+// TODO TEST add a test that starts the SDK in a docker container and checks some simple cmds
+// TODO DEPS make JitPack jars for catalog and NMS and use them
+
+repositories {
+	mavenCentral()
+	maven {  // for syslog4j
+		name = "Clojars"
+		url = "https://repo.clojars.org/"
+	}
+}
+
+task buildGitCommitFile {
+	doLast {
+		def commitId = grgit.head().id
+		// is there a variable for builddir/classes/java/main?
+		file("$buildDir/classes/java/main/us/kbase/mobu/gitcommit").text = commitId
+	}
+}
+
+compileJava {
+	options.release = 11
+	finalizedBy buildGitCommitFile
+}
+
+task pythonTest(type: Exec) {
+	description = 'Runs Python tests using pytest'
+	environment 'PYTHONPATH', './src/java/us/kbase/templates'
+	commandLine 'pytest',
+		'--cov=authclient', '--cov-report=xml', '--cov-report=html', 'test_scripts/py_module_tests'
+	doFirst {
+		println "Running: ${commandLine.join(' ')}"
+	}
+}
+
+test {
+	dependsOn pythonTest
+	
+	maxHeapSize = "3G"
+	testLogging {
+		exceptionFormat = 'full'
+		showStandardStreams = true
+	}
+	finalizedBy jacocoTestReport
+}
+
+jacocoTestReport {
+	reports {
+		xml.required = true
+		csv.required = true
+	}
+	
+	// Exclude specific directories or files from coverage
+	afterEvaluate {
+		classDirectories.setFrom(
+			files(classDirectories.files.collect {
+				fileTree(dir: it, exclude: [
+					"**/fraser/neil/**",
+					"**/common/executionengine/**",
+					"**/common/service/**",
+					"**/catalog/**",
+					"**/narrativemethodstore/**",
+				])
+			})
+		)
+	}
+}
+
+// Custom java project layout
+sourceSets {
+	main {
+		java {
+			srcDirs = ["src/java/"]
+			exclude "**/test/**"
+		}
+		resources {
+			srcDirs = ["src/java/"]
+			include "**/gitcommit"
+			include "**/*.css"
+			include "**/*.properties"
+			include "**/*.py"
+			exclude "**/test**"
+		}
+	}
+	test {
+		java {
+			srcDirs = ["src/java/"]
+			include '**/test/**'
+		}
+		resources {
+			srcDirs = ["src/java/"]
+			include "**/test/**/*.properties"
+			include "**/test/**/*.css"
+			include "**/test/**/*.spec"
+			include "**/test/**/*.html"
+		}
+	}
+}
+
+configurations {
+	// can't directly access testImplementation, so extend and access
+	testimpl.extendsFrom testImplementation
+}
+
+def fromURL = { url, name ->
```

Comment:
is this the same as the function used in the workspace gradle file?

Reply:
This function used to be in the workspace but was only temporary until it was switched to Jitpacked jars. The same thing is going to happen here (at least that's the plan) and this function will be removed